Source/ChilliSource/Rendering/Base/Renderer.cpp

```diff
@@ -43,55 +66,68 @@ namespace ChilliSource
-        
-        return RenderSnapshot(allocator, resolution, clearColour, renderCamera);
+        return RenderSnapshot(renderTarget, resolution, clearColour, renderCamera);
     }
     
     //------------------------------------------------------------------------------
-    void Renderer::ProcessRenderSnapshot(RenderSnapshot renderSnapshot) noexcept
+    void Renderer::ProcessRenderSnapshots(IAllocator*&& frameAllocator, RenderSnapshot&& mainRenderSnapshot, std::vector<RenderSnapshot>&& offscreenRenderSnapshots) noexcept
     {
         WaitThenStartRenderPrep();
         
-        m_currentSnapshot = std::move(renderSnapshot);
+        m_currentMainSnapshot = std::move(mainRenderSnapshot);
+        m_currentOffscreenSnapshots = std::move(offscreenRenderSnapshots);
         
         auto taskScheduler = Application::Get()->GetTaskScheduler();
+        
         taskScheduler->ScheduleTask(TaskType::k_small, [=](const TaskContext& taskContext)
         {
-            auto resolution = m_currentSnapshot.GetResolution();
-            auto clearColour = m_currentSnapshot.GetClearColour();
-            auto renderCamera = m_currentSnapshot.GetRenderCamera();
-            auto renderAmbientLights = m_currentSnapshot.ClaimAmbientRenderLights();
-            auto renderDirectionalLights = m_currentSnapshot.ClaimDirectionalRenderLights();
-            auto renderPointLights = m_currentSnapshot.ClaimPointRenderLights();
-            auto renderObjects = m_currentSnapshot.ClaimRenderObjects();
-            auto preRenderCommandList = m_currentSnapshot.ClaimPreRenderCommandList();
-            auto postRenderCommandList = m_currentSnapshot.ClaimPostRenderCommandList();
-            auto renderFrameData = m_currentSnapshot.ClaimRenderFrameData();
+            std::vector<RenderFrame> renderFrames(m_currentOffscreenSnapshots.size());
+            std::vector<RenderFrameData> renderFramesData(m_currentOffscreenSnapshots.size());
+            
+            for(auto i=0; i<m_currentOffscreenSnapshots.size(); ++i)
+            {
+                auto renderFrameData = m_currentOffscreenSnapshots[i].ClaimRenderFrameData();
+                renderFramesData.push_back(std::move(renderFrameData));
+                
+                auto renderFrame = CompileRenderFrame(m_currentOffscreenSnapshots[i]);
+                renderFrames.push_back(std::move(renderFrame));
+            }
             
-            auto renderFrame = RenderFrameCompiler::CompileRenderFrame(resolution, clearColour, renderCamera, renderAmbientLights, renderDirectionalLights, renderPointLights, renderObjects);
-            auto targetRenderPassGroups = m_renderPassCompiler->CompileTargetRenderPassGroups(taskContext, renderFrame);
-            auto renderCommandBuffer = RenderCommandCompiler::CompileRenderCommands(taskContext, targetRenderPassGroups, std::move(preRenderCommandList), std::move(postRenderCommandList), std::move(renderFrameData));
```

Comment:
There is no need to move the frames here, just pass references. This method doesn't exit until all of the tasks spawned tasks have completed meaning it's safe to pass references to stack objects.